### DISCUSSION TITLE
L1D and L2  cache config for RTX3070 seems fishy!!

### DISCUSSION THREAD

**Discussion by sroy729** [accel-sim/accel-sim-framework#323] [Q&A] [answered]
The config for L1D RTX3070 in `gpgpu-sim_distribution` shows the number of ways is 256, which is quite high.
 Can I get a clarification on this 
 line 95: [config](https://github.com/accel-sim/gpgpu-sim_distribution/blob/6aa7ed16ed4c244bebaf8942f8666bbd94a2c757/configs/tested-cfgs/SM86_RTX3070/gpgpusim.config)
 
 The config for L2 in gpgpusim.config 
 ```
 -gpgpu_cache:dl2 S:64:128:16,L:B:m:L:P,A:192:4,32:0,32
```
so the L2 cache size turn out to be `64x128x16 = 128KiB` according to the config above
but in the [white paper](https://www.nvidia.com/content/PDF/nvidia-ampere-ga-102-gpu-architecture-whitepaper-v2.1.pdf) Table 10 Appendix A the RTX3070 config shows L2 cache size of 4096KiB. 
Can you help me understand this? Does it have something to do with banking and memory controllers, as the size specified is 128x32?

**Comment by JRPan** [comment, marked as answer]
dl1: 4,128,256 = 128kB. You can play with set/assoc to see how it correlates, as long as the total size is 128kB. 

for dl2:
```
-gpgpu_n_mem 16
-gpgpu_n_sub_partition_per_mchannel 2
```
There are 16 memory controllers (or memory channels). And each controller has 2 L2 banks (sub partitions). So in total, 32 banks. 
Each bank is 128KB as you said, 128K*32 = 4MB.
